Adding single week paging

dominikzarzecki opened the pull request:
Added support for different paging modes (monthly and weekly).
CalendarPagerView and CalendarPagerAdapter have most code migrated from MonthView and MonthPagerAdapter, so it can be reused in WeekView and WeekPagerAdapter.

ekchang commented:
Looks good. As a part of this PR, please update `TitleChanger.change(currentMonth)` to also exit early if `currentMonth.getMonth() == previousMonth.getMonth()` (add as part of line 54)

This will prevent the month title formatter from running the animation during Week Mode if you are paging between weeks within the same month.

dominikzarzecki commented:
Changed the requested line. There was a bug with calculating the first week day, and this caused collapsing to the wrong week. This is now fixed and should be covered by the added test.

ekchang commented:
Well done. Thanks.
